scheduler/scheduler.go

```diff
@@ -337,17 +342,73 @@ func (s *scheduler) createTask(sch schedule.Schedule, wfMap *wmap.WorkflowMap, s
 		return nil, te
 	}
 
+	// subscribedPluginAsserts includes rules that need to be evaluated once we
+	// have mapped the metrics to specific collector plugins.  Examples include
+	// asserting that streaming tasks don't reference non-streaming collectors.
+	subscribedPluginAsserts := []core.SubscribedPluginAssert{}
 	// Group dependencies by the node they live on
 	// and validate them.
 	depGroups := getWorkflowPlugins(wf.processNodes, wf.publishNodes, wf.metrics)
 	for k, group := range depGroups {
+
+		// populate subscribedPluginAsserts
+		switch sch.(type) {
+		case *schedule.StreamingSchedule:
+			// assert no non-streaming plugins
+			subscribedPluginAsserts = append(subscribedPluginAsserts, func(plugins []core.SubscribedPlugin) serror.SnapError {
+				for _, plg := range plugins {
+					if plg.TypeName() != plugin.StreamCollectorPluginType.String() {
+						return serror.New(
+							ErrPluginIncompatibleWithScheduleType,
+							map[string]interface{}{
+								"schedule_type": fmt.Sprintf("%T", sch),
+								"plugin_name":   plg.Name(),
+								"plugin_type":   plg.TypeName(),
+							},
+						)
+					}
+				}
+				return nil
+			})
+			// assert only a single streaming plugin
+			subscribedPluginAsserts = append(subscribedPluginAsserts, func(plugins []core.SubscribedPlugin) serror.SnapError {
+				if len(plugins) > 1 {
+					return serror.New(
+						ErrMultipleStreamingPlugins,
+						map[string]interface{}{
+							"schedule_type":     fmt.Sprintf("%T", sch),
+							"num_of_collectors": len(plugins),
+						},
+					)
+				}
+				return nil
+			})
+		default:
+			// assert no streaming plugins
+			subscribedPluginAsserts = append(subscribedPluginAsserts, func(plugins []core.SubscribedPlugin) serror.SnapError {
+				for _, plg := range plugins {
+					if plg.TypeName() == plugin.StreamCollectorPluginType.String() {
+						return serror.New(
+							ErrPluginIncompatibleWithScheduleType,
+							map[string]interface{}{
+								"schedule_type": fmt.Sprintf("%T", sch),
+								"plugin_name":   plg.Name(),
+								"plugin_type":   plg.TypeName(),
```

Comment:
I fully agree, this is the issue which should be addressed outside the scope of this PR. I  created GH issue to address that in this way - https://github.com/intelsdi-x/snap/issues/1654